### PR TITLE
NAS-111194 / 21.08 / document return types in failover plugin(s)

### PR DESCRIPTION
**Pull request by yocalebo**
- document return types in failover plugin(s)
- while here, remove the `_linux.py` suffix for the failover related files

### CODE REVIEW

File: src/middlewared/middlewared/plugins/failover_/remote.py [hunk 1]
```diff
@@ -224,6 +224,7 @@ async def remote_ip(self):
             Any('callback'),
         ),
     )
+    @returns(Any())
```
Comment (sonicaj) [inline src/middlewared/middlewared/plugins/failover_/remote.py:227]: `null=False` by default, can you please set that to `true` ?

File: src/middlewared/middlewared/plugins/failover.py [hunk 2]
```diff
@@ -90,6 +90,12 @@ async def failover_extend(self, data):
         Bool('master', null=True),
         update=True,
     ))
+    @returns(Dict(
+        Int('id'),
+        Bool('disabled'),
+        Int('timeout'),
```
Comment (sonicaj) [inline src/middlewared/middlewared/plugins/failover.py:96]: I suppose these are always going to be present ? We should add `required=True` in that case

**Review by sonicaj — CHANGES_REQUESTED** [review]
Have we documented `failover.config` ?

File: src/middlewared/middlewared/plugins/failover.py [hunk 3]
```diff
@@ -78,6 +78,14 @@ class Config:
         datastore_extend = 'failover.failover_extend'
         cli_namespace = 'system.failover'
 
+    ENTRY = Dict(
+        'failover_entry',
+        Int('id', required=True),
+        Bool('disabled', required=True),
+        Int('timeout', required=True),
+        Bool('master', required=True),
```
Comment (sonicaj) [inline src/middlewared/middlewared/plugins/failover.py:86]: `master` can be `null` i assume ?

**Review by sonicaj — COMMENTED** [review]
Have we deleted all the _linux files ?

File: src/middlewared/middlewared/plugins/failover.py (hunk 3 above)
Comment (yocalebo) [reply src/middlewared/middlewared/plugins/failover.py:86]: No, it always exists in the db as a boolean.

Comment (sonicaj) [reply src/middlewared/middlewared/plugins/failover.py:86]: Why do we allow it to be null on update method ?

Comment (yocalebo) [reply src/middlewared/middlewared/plugins/failover.py:86]: If it's null, then we figure out which node is to become (or stay) the master node.

**Review by sonicaj — APPROVED** [review]
Thanks!

**Pull request merged**